### PR TITLE
Support combining of entities with etched/glossy/foil/non status.

### PR DESCRIPTION
**Pull request by ZeldaZach**
This will reduce the number of duplicate entries, and help simplify the card model. This also adds referral URL support.

ADD
- card.finishes
- card.identifiers.tcgplayer_etched_product_id
- card.purchase_urls.tcgplayer_etched

DEPRECATE
- has_foil (Will be removed in 5.2.0)
- has_non_foil (Will be removed in 5.2.0)

[CMR.json.txt](https://github.com/mtgjson/mtgjson/files/7132572/CMR.json.txt)
[H1R.json.txt](https://github.com/mtgjson/mtgjson/files/7132561/H1R.json.txt)

### CODE REVIEW

File: mtgjson5/providers/tcgplayer.py [hunk 1]
```diff
@@ -257,9 +257,7 @@ def generate_mtgjson_sealed_product_objects(
                 sealed_product.release_date = sealed_product.release_date[0:10]
             sealed_product.raw_purchase_urls[
                 "tcgplayer"
-            ] = "https://shop.tcgplayer.com/product/productsearch?id={}&utm_campaign=affiliate&utm_medium=api&utm_source=mtgjson".format(
-                sealed_product.identifiers.tcgplayer_product_id
-            )
+            ] = f"https://shop.tcgplayer.com/product/productsearch?id={sealed_product.identifiers.tcgplayer_product_id}&utm_campaign=affiliate&utm_medium=api&utm_source=mtgjson"
```
Comment (kodawah) [inline mtgjson5/providers/tcgplayer.py:260]: this seems an unrelated cosmetic change

File: mtgjson5/set_builder.py [hunk 2]
```diff
@@ -713,12 +713,12 @@ def build_mtgjson_card(
...
     mtgjson_card.edhrec_rank = scryfall_object.get("edhrec_rank")
+    mtgjson_card.finishes = scryfall_object.get("finishes")
     mtgjson_card.frame_effects = scryfall_object.get("frame_effects", "")
     mtgjson_card.frame_version = scryfall_object.get("frame", "")
     mtgjson_card.hand = scryfall_object.get("hand_modifier")
-    mtgjson_card.has_foil = scryfall_object.get("foil")
-    mtgjson_card.has_non_foil = scryfall_object.get("nonfoil")
-
+    mtgjson_card.has_foil = "foil" in scryfall_object.get("finishes", [])
+    mtgjson_card.has_non_foil = "nonfoil" in scryfall_object.get("finishes", [])
```
Comment (kodawah) [inline mtgjson5/set_builder.py:723]: for compatibility you should check for `glossy` too

File: mtgjson5/providers/tcgplayer.py (hunk 1 above)
Comment (ZeldaZach) [reply mtgjson5/providers/tcgplayer.py:260]: Yes, yes it is. Small change, no relevance to anything.

File: mtgjson5/set_builder.py (hunk 2 above)
Comment (ZeldaZach) [reply mtgjson5/set_builder.py:723]: There won't be a `hasGlossy` as we're deprecating foil/nonfoil top level (5.2.0 removal). Use finishes instead

Comment (kodawah) [reply mtgjson5/set_builder.py:723]: i'm saying that glossy cards that used to be with `hasfloil` set, won't be anybor with this simple check

this is a break that is easily avoidable

File: mtgjson5/providers/tcgplayer.py (hunk 1 above)
Comment (kodawah) [reply mtgjson5/providers/tcgplayer.py:260]: it just makes review harder 😅

File: mtgjson5/set_builder.py (hunk 2 above)
Comment (ZeldaZach) [reply mtgjson5/set_builder.py:723]: Ahh yep, this has been fixed. :D

**Review by kodawah — APPROVED** [review]
